### PR TITLE
set target modal

### PR DESCRIPTION
**Pull request by AnosikeChidinma**
**Description of PR**

Designed the set target modal.

---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [x] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ yes] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [ ] Does the endpoint return the correct success/error format?

---

**Screenshots**  
_add relevant screenshots here_

Screenshot 2024-02-02 at 3 06 17 PM

---

**Test Coverage**  
_add relevant screenshots here_

### CODE REVIEW

**Comment by KristofaJosh** [comment]
Task linked: [CU-86bwmcruu Set a target](https://app.clickup.com/t/86bwmcruu)

**Review by pelzman — COMMENTED** [review]
image


it better you render this using map method, so you don't repeat yourself

**Review by pelzman — COMMENTED** [review]
also onClick each Text , setVisible(false), so that the dropdown close immediately any text is selected.....
like  onClick={()=>{
handleFrequencySelection(....)
setVisible(false)
}}

**Comment by AnosikeChidinma** [comment]
> also onClick each Text , setVisible(false), so that the dropdown close immediately any text is selected..... like onClick={()=>{ handleFrequencySelection(....) setVisible(false) }}

That was already handled in the code, once a text is clicked, the dropdown closes.
Thank you for your review.

**Review by Elyzsium — APPROVED** [review]

File: apps/client/src/components/compounds/SetTarget/ClosecloseIcon.svg
Comment (KristofaJosh) [inline apps/client/src/components/compounds/SetTarget/ClosecloseIcon.svg:1]: I believe we already have closeIcon somewhere in our codebase, please use that.

File: apps/client/src/components/compounds/SetTarget/index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const handleFrequencySelection = (frequency: string) => {
+    setSelectedFrequency(frequency)
+    setVisible(false)
+  }
+
+  return (
+    <div>
+      <Card className={styles.setTargetContainer}>
+        <div className={styles.setTargetContainerCloseIcon} onClick={onClose}>
+          <CloseIcon />
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/SetTarget/index.tsx:27]: should be wrapped with a button tag

File: apps/client/src/components/compounds/SetTarget/index.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                  <Text
+                    content="Weekly"
+                    onClick={() => handleFrequencySelection('Weekly')}
+                  />
+                  <Text
+                    content="Monthly"
+                    onClick={() => handleFrequencySelection('Monthly')}
+                  />
+                </div>
+              )}
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/SetTarget/index.tsx:68]: The ability to translate UI is very important.
This is a select/dropdown UI component, why write the code like this?

We have a select component and you installed react-select, is there a reason you aren't using any of them here?

File: apps/client/src/components/compounds/SetTarget/index.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            </div>
+            <Input label={'Start Date'} type={'date'} />
+            <Input
+              label={'Withdrawal Date'}
+              placeholder={'Pick your date'}
+              type={'date'}
+            />
+            <Button text="Submit" type="submit" />
+          </form>
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/SetTarget/index.tsx:77]: Your form is not validated

File: apps/client/src/components/compounds/SetTarget/setTarget.module.scss [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      right: -2%;
+      bottom: -108%;
+      z-index: 40;
+      line-height: 32.4px;
+      padding: 10px 10px;
+      border-radius: 10px;
+    }
+  }
+  @media (max-width: 425px) {
+    & {
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/SetTarget/setTarget.module.scss:56]: ??

**Review by KristofaJosh — CHANGES_REQUESTED** [review]
Please make those changes requested

**Comment by AnosikeChidinma** [comment]
> image
> 
> it better you render this using map method, so you don't repeat yourself


> image
> 
> it better you render this using map method, so you don't repeat yourself

Noted

File: apps/client/src/components/compounds/SetTarget/ClosecloseIcon.svg
Comment (AnosikeChidinma) [reply apps/client/src/components/compounds/SetTarget/ClosecloseIcon.svg:1]: Ok sir

File: apps/client/src/components/compounds/SetTarget/index.tsx (hunk 1 above)
Comment (AnosikeChidinma) [reply apps/client/src/components/compounds/SetTarget/index.tsx:27]: Noted

File: apps/client/src/components/compounds/SetTarget/index.tsx (hunk 2 above)
Comment (AnosikeChidinma) [reply apps/client/src/components/compounds/SetTarget/index.tsx:68]: The select component wasn't built in a way that could allow/permit a custom feature and react-select select was allowing my custom styling especially the width to over-ride the default width of react-select that was why i didn't use either of them but didn't remember to remove the library from the project.

File: apps/client/src/components/compounds/SetTarget/index.tsx (hunk 3 above)
Comment (AnosikeChidinma) [reply apps/client/src/components/compounds/SetTarget/index.tsx:77]: I'll add validation to it now

File: apps/client/src/components/compounds/SetTarget/setTarget.module.scss (hunk 4 above)
Comment (AnosikeChidinma) [reply apps/client/src/components/compounds/SetTarget/setTarget.module.scss:56]: i was adding responsiveness to it

& {: The "&" symbol refers to the parent selector. In this case, it represents the main container or element when the screen width matches the media query condition.

File: apps/client/src/components/compounds/SetTarget/index.tsx (hunk 2 above)
Comment (KristofaJosh) [reply apps/client/src/components/compounds/SetTarget/index.tsx:68]: regardless, your code representation of the UI is wrong.

react select is very customizable, I believe it's possible to style the width 
https://react-select.com/styles

File: apps/client/src/components/compounds/SetTarget/setTarget.module.scss (hunk 4 above)
Comment (KristofaJosh) [reply apps/client/src/components/compounds/SetTarget/setTarget.module.scss:56]: you could do this instead:

```
@media (max-width: 425px){
   width: 100%;
      &Input {
        width: 100%;
      }
      &Text {
        &Create {
          width: 100%;
        }
      }
    }
    ```

File: apps/client/src/components/compounds/SetTarget/index.tsx (hunk 2 above)
Comment (KristofaJosh) [reply apps/client/src/components/compounds/SetTarget/index.tsx:68]: besides, we already have a Select component in the codebase, if you can't modify any of the external libraries, then use our default

File: apps/client/src/components/compounds/SetTarget/index.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+              />
+              {visible && (
+                <div className={styles.setTargetContainerInputHidden}>
+                  {frequencyOptions.map((frequency) => (
+                    <Text
+                      key={frequency}
+                      content={frequency}
+                      onClick={() => handleFrequencySelection(frequency)}
+                    />
+                  ))}
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/SetTarget/index.tsx:111]: You should use the select input

**Review by KristofaJosh — CHANGES_REQUESTED** [review]

File: apps/client/src/components/compounds/SetTarget/index.tsx (hunk 3 above)
Comment (AnosikeChidinma) [reply apps/client/src/components/compounds/SetTarget/index.tsx]: I've validated the form now.

File: apps/client/src/components/compounds/SetTarget/index.tsx (hunk 5 above)
Comment (AnosikeChidinma) [reply apps/client/src/components/compounds/SetTarget/index.tsx:111]: Alright sir, I'lll do that right away.

File: apps/client/src/components/compounds/SavingsHeader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div className={styles.goalsHeader}>
+      <Text content={'My Goals'} size={'Subtext'} />
+      <Modal
+        modalContentClassName={styles.modalContent}
+        disableOutsideClose
+        renderOnOpen={({ onOpen }) => (
+          <Link onClick={onOpen} to={''}>
+            Add New Goal
+          </Link>
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/SavingsHeader/index.tsx:16]: This should be a button.
style it to match accordingly

**Review by pelzman — APPROVED** [review]

**Review by KristofaJosh — APPROVED** [review]

**Pull request merged**